### PR TITLE
Upgrade jQuery to 3.5.0, remove jQuery-Migrate, fix deprecations

### LINKED ISSUE #9372: Remove jQuery Migrate
[jQuery Migrate](https://github.com/jquery/jquery-migrate) was added in https://github.com/go-gitea/gitea/pull/7425 for the jQuery 3.0 migration but it's not something that's meant to be shipped permanently. From their README:

> jQuery Migrate makes this easier, by restoring the APIs that were removed ... That way you can spot and fix what otherwise would have been errors, until you no longer need jQuery Migrate and can remove it.

@techknowlogick were there any specific issues that lead to you integrating it or was it only added  "just in case"?

### PR DESCRIPTION
**Pull request by silverwind**
Upgrade jQuery to 3.5.0, remove jQuery-Migrate

- Removed jQuery-Migrate as it seems to no longer be needed.
- Removed a dead code section.
- Fixed some deprecations detected by jQuery-Migrate.

Ref: https://jquery.com/upgrade-guide/3.5/
Ref: https://github.com/gogs/gogs/search?q=repo-name-change-prompt
Ref: https://github.com/go-gitea/gitea/search?q=repo-name-change-prompt
Fixes: https://github.com/go-gitea/gitea/issues/9372

### CODE REVIEW

**Comment by 6543** [comment]
@silverwind bad news: dropzone do not work anymore ... looks like it still depend on migrate :(

**Comment by silverwind** [comment]
It did work for me, at least adding/removing attachment. What was not working for you?

**Comment by 6543** [comment]
adding attachments to a new created issue

**Comment by silverwind** [comment]
Will check. I did not yet run with unmuted jquery-Migrate which should highlight all the issue in the console, but it's a bit cumbersome in general because of SimpleMDE console spam.

**Comment by guillep2k** [comment]
Dropzone itself claims to have no external dependencies, even jQuery... 🤔 Maybe it's the way we use it?

**Comment by silverwind** [comment]
Yeah, something seems fishy as we don't use the jQuery-variant of Dropzone initialization. I think the old 4.x version of Dropzone may have had a jQuery dependency, but the current version we use doesn't.

**Comment by silverwind** [comment]
@6543 I played around a bit with dropzone and I can't find any difference before or after this PR. 

Adding attachments works, there's just the non-obvious case when the attachment exceeds the file size limit (4mb), hitting save will still save the comment but the attachment is gone from the comment. I think it might better to just prevent save in that case. Maybe this was what you were seeing?

**Comment by 6543** [comment]
no I could not use it att all - I'll recheck have a clean test env now

**Review by 6543 — APPROVED** [review]
works fine now

`¯\_(ツ)_/¯` why it didn't before

**Comment by silverwind** [comment]
I guess many such mysteries can be attributed to a stale service worker cache, if you have that enabled.

**Comment by 6543** [comment]
Already disabled for exact that reason

**Review by guillep2k — APPROVED** [review]

**Comment by silverwind** [comment]
Unmuted migrate reveals a few deprecations, some in our code. I will work on eliminating them, don't land yet.

```
JQMIGRATE: jQuery.expr[':'] is deprecated; use jQuery.expr.pseudos
JQMIGRATE: jQuery.fn.click() event shorthand is deprecated
JQMIGRATE: jQuery.trim is deprecated; use String.prototype.trim
JQMIGRATE: jQuery.isFunction() is deprecated
JQMIGRATE: jQuery.isWindow() is deprecated
JQMIGRATE: jQuery.fn.bind() is deprecated
JQMIGRATE: jQuery.fn.submit() event shorthand is deprecated
JQMIGRATE: jQuery.fn.unbind() is deprecated
JQMIGRATE: jQuery.fn.keypress() event shorthand is deprecated
JQMIGRATE: jQuery.fn.keyup() event shorthand is deprecated
JQMIGRATE: jQuery.fn.change() event shorthand is deprecated
JQMIGRATE: jQuery.fn.keydown() event shorthand is deprecated

**Comment by silverwind** [comment]
Fixed all deprecations in our code. Some still remain, but these are in dependencies or vendor files.

File: web_src/js/index.js [hunk 1]
```diff
@@ -728,17 +728,8 @@ async function initRepository() {
 
   // Options
   if ($('.repository.settings.options').length > 0) {
-    $('#repo_name').keyup(function () {
```
Comment (6543) [inline web_src/js/index.js:731]: @silverwind can we just delete this?

Comment (silverwind) [reply web_src/js/index.js:731]: Yes, it's dead code, the `#repo-name-change-prompt` does not exist:

Ref: https://github.com/gogs/gogs/search?q=repo-name-change-prompt
Ref: https://github.com/go-gitea/gitea/search?q=repo-name-change-prompt

Comment (silverwind) [reply web_src/js/index.js:731]: I did not check history why this prompt was removed, but feel free to dig into it :)

Comment (6543) [reply web_src/js/index.js:731]: 🔪 it

**Review by 6543 — APPROVED** [review]

**Comment by techknowlogick** [comment]
ping LG-TM

**Pull request merged**